tests/TestCase/ORM/TableTest.php

```diff
@@ -4515,6 +4516,349 @@ public function testReplacelinksBelongsToManyWithJoint()
     }
 
     /**
+     * Tests that options are being passed through to the internal table method calls.
+     *
+     * @return void
+     */
+    public function testOptionsBeingPassedToImplicitBelongsToManyDeletesUsingSaveReplace()
+    {
+        $articles = TableRegistry::get('Articles');
+
+        $tags = $articles->belongsToMany('Tags');
+        $tags->saveStrategy(BelongsToMany::SAVE_REPLACE);
+        $tags->dependent(true);
+        $tags->cascadeCallbacks(true);
+
+        $actualOptions = null;
+        $tags->junction()->eventManager()->on(
+            'Model.beforeDelete',
+            function (Event $event, Entity $entity, ArrayObject $options) use (&$actualOptions) {
+                $actualOptions = $options->getArrayCopy();
```

Comment:
Are you sure that it's safe to assume that the callbacks are being invoked? I've had some bad experience with that in the past, where tests passed because of the events not being triggered (for whatever reason).